### PR TITLE
pkg_version_info "==" comparison sign

### PR DESCRIPTION
**Pull request by KryeKuzhinieri**
This pull request is based on the following issue https://github.com/r-lib/rlang/issues/1469. 
As explained in the issue, the goal is to allow `==` comparison when `pkg_version_info` function is called. I am not sure if this will be supported by the `rlang` team, however, such addition would be more than welcome. 

Besides, I have added the necessary tests and tested the install on my personal machine (`Ubuntu 22.04`, R versions (`4.1.3` and `4.2.1`)) and it worked fine. 

If this PR does not suit the guidelines of the rlang team, I can modify it as per your needs.

### CODE REVIEW

**Comment by lionel-** [comment]
Can you add a NEWS bullet and tag yourself and your issue please?

**Comment by KryeKuzhinieri** [comment]
> Can you add a NEWS bullet and tag yourself and your issue please?

This has been completed @lionel-

File: NEWS.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
   interfaces with `obj_type_friendly()` from `compat-obj-type.R` via a
   C callable.
 
+* The `pkg_version_info()` function now allows `==` for package version comparison. ([#1469](https://github.com/r-lib/rlang/issues/1469), [@kryekuzhinieri](https://github.com/KryeKuzhinieri))
```
Comment (lionel-) [inline NEWS.md:12]: The bullet should be under the development version header.

Comment (KryeKuzhinieri) [reply NEWS.md:12]: Shall I add it under `1.0.5.9000 `or increment the development version to `1.0.6`?

Comment (lionel-) [reply NEWS.md:12]: Just add the bullet under the very first header in the file, thanks!

**Comment by lionel-** [comment]
Thanks!

File: NEWS.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
 # rlang (development version)
 
-# rlang 1.0.5
+* The `pkg_version_info()` function now allows `==` for package
```
Comment (KryeKuzhinieri) [inline NEWS.md:3]: I see what you meant now. Thanks

**Pull request merged**